File: 2021/May/Day 9/self-dividing-number.py
# Source:https://leetcode.com/problems/self-dividing-numbers/submissions/
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        lists =[i for i in range(left,right+1) if self.isSelfDividing(i)]
        return lists
            
    def isSelfDividing(self,num):
        n = num;
        while(n):
            digit = int(n%10);
            if(digit==0 or not num%digit==0):
                return False
            n = int(n/10);
        return True

File: 2022/March/Day 28/selfDividingNumber.cpp
// Source : https://leetcode.com/problems/self-dividing-numbers/submissions/
class Solution {
public:

    bool selfDiv(int i)
    {
        int copy=i;
        int d;

        while(i)
        {
            d = i%10;
            if(d==0){return false;}
            else if(copy%(d)!=0)
            {
                return false;
            }
            i/=10;
        }
        return true;


    }
    vector<int> selfDividingNumbers(int left, int right) {

        vector<int> out;
        for(int i=left;i<=right;i++)
        {
            if(selfDiv(i))
            {
                out.insert(out.end(),i);
            }

        }
        return out;
    }
};
